src/appleseed/foundation/array/array.cpp

```diff
@@ -44,7 +44,109 @@ namespace foundation
 // Array class implementation.
 //
 
-Array::Concept::~Concept()
+template <typename T>
+Array::Concept* Array::Model<T>::copy() const
```

Comment:
Mark these methods as inline? Not sure it makes much of a difference...

Reply:
Probably not. All these methods are virtual.